ldaptive/src/main/java/org/ccci/idm/user/ldaptive/dao/LdaptiveUserDao.java

```diff
@@ -632,29 +641,61 @@ private int enqueueGroupsByFilter(@Nonnull final Collection<Group> groups, @Null
         }
     }
 
-    private void modifyGroupMembership(User user, Group group, AttributeModificationType attributeModificationType)
-            throws DaoException {
+    private void modifyGroupMembership(@Nonnull final AttributeModificationType type, @Nonnull final User user,
+                                       @Nonnull final Group group, final boolean updateSecurity) throws DaoException {
         Connection conn = null;
         try {
             conn = this.connectionFactory.getConnection();
             conn.open();
 
-            String userDn = this.userMapper.mapDn(user);
+            final String userDn = userMapper.mapDn(user);
             final String groupDn = DnUtils.toString(group);
+            final ResultCode suppressedError =
+                    type == AttributeModificationType.ADD ? ResultCode.ATTRIBUTE_OR_VALUE_EXISTS :
+                            type == AttributeModificationType.REMOVE ? ResultCode.NO_SUCH_ATTRIBUTE :
+                                    null;
 
             // modify user entry
-            modifyEntry(conn, userDn, attributeModificationType, groupDn, LDAP_ATTR_GROUPS);
+            modifyEntrySuppressed(conn, userDn, type, LDAP_ATTR_GROUPS, groupDn, suppressedError);
+            if (updateSecurity) {
+                modifyEntrySuppressed(conn, userDn, type, LDAP_ATTR_SECURITY_EQUALS, groupDn, suppressedError);
+            }
 
             // modify group entry
-            modifyEntry(conn, groupDn, attributeModificationType, userDn, LDAP_ATTR_MEMBER);
-
+            modifyEntrySuppressed(conn, groupDn, type, LDAP_ATTR_MEMBER, userDn, suppressedError);
+            if (updateSecurity) {
+                modifyEntrySuppressed(conn, groupDn, type, LDAP_ATTR_EQUIVALENT_TO_ME, userDn, suppressedError);
+            }
         } catch (final LdapException e) {
             throw convertLdapException(e);
         } finally {
             LdapUtils.closeConnection(conn);
         }
     }
 
+    private boolean modifyEntrySuppressed(@Nonnull final Connection conn, @Nonnull final String dn,
```

Comment:
Would it not be easier to simply encapsulate suppression logic within `modifyEntrySuppressed`? It seems rather expected behavior.